### PR TITLE
ENH: Added resources support to pod config

### LINKED ISSUE #98: Add resource requests and limits to benchmarks
https://github.com/xridge/kubestone/blob/6da07a4d08128cd8cbf15f084b7d9cf17af81df8/controllers/sysbench/job.go#L68-L73

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 6da07a4d08128cd8cbf15f084b7d9cf17af81df8 when #71 was merged. cc @apollolemonte.

### PR DESCRIPTION
**Pull request by julio-santiesteban**
Fixes #98

### CODE REVIEW

File: controllers/common/job_test.go [hunk 1]
```diff
@@ -127,5 +146,24 @@ var _ = Describe("perf job", func() {
...
+			It("should request the given memory", func() {
+				Expect(job.Spec.Template.Spec.Containers[0].Resources.Requests.Memory()).To(
+					BeEquivalentTo(podConfig.Resources.Requests.Memory()))
+			})
+			It("should limit to the given CPU", func() {
+				Expect(job.Spec.Template.Spec.Containers[0].Resources.Limits.Cpu()).To(
+					BeEquivalentTo(podConfig.Resources.Limits.Cpu()))
+			})
+			It("should limit to the given memory", func() {
+				Expect(job.Spec.Template.Spec.Containers[0].Resources.Limits.Cpu()).To(
```
Comment (otto-von-bivouac) [inline controllers/common/job_test.go]: Did you mean to check for memory here?

File: controllers/common/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestCommonController(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Fio Common Suite")
```
Comment (otto-von-bivouac) [inline controllers/common/suite_test.go]: Common Suite

File: controllers/iperf3/serverdeployment_test.go [hunk 3]
```diff
@@ -159,5 +170,24 @@ var _ = Describe("Server Deployment", func() {
...
+			It("should request the given memory", func() {
+				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Requests.Memory()).To(
+					BeEquivalentTo(cr.Spec.ServerConfiguration.Resources.Requests.Memory()))
+			})
+			It("should limit to the given CPU", func() {
+				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits.Cpu()).To(
+					BeEquivalentTo(cr.Spec.ServerConfiguration.Resources.Limits.Cpu()))
+			})
+			It("should limit to the given memory", func() {
+				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits.Cpu()).To(
```
Comment (otto-von-bivouac) [inline controllers/iperf3/serverdeployment_test.go]: Should be memory

**Comment by otto-von-bivouac** [comment]
As always, looks awesome!
Once the minor typos are fixed this is good to merge.

File: controllers/common/job_test.go (hunk 1 above)
Comment (julio-santiesteban) [reply controllers/common/job_test.go]: yes :)

**Pull request merged**